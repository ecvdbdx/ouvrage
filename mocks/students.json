[
  {
    "username": "epatrois",
    "first_name": "Manu",
    "last_name": "Patrois",
    "id": "2"
  },
  {
    "id": "5",
    "username": "mdewaele",
    "first_name": "Maxence",
    "last_name": "De Waele",
    "email": "maxencedewaele@gmail.com",
    "birthdate": "838921463",
    "avatar": "https://cdn.pixabay.com/photo/2015/10/05/22/37/blank-profile-picture-973460_960_720.png"

  },
  {
    "id": "8",
    "username": "rodrigodelanoche",
    "first_name": "Vincent",
    "last_name": "Rodriguez",
    "email": "vincentrodrigu@hotmail.fr",
    "birthdate": "14/05/1413",
    "avatar": "http://etc.usf.edu/clipart/5500/5596/columbus_14_md.gif"
  },
  {
    "id": "6",
    "username": "erozis",
    "first_name": "Emilie",
    "last_name": "Rozis",
    "email": "emilie.40@hotmail.com"
  },
  {
    "first_name": "Hubert",
    "username": "hmoncenis",
    "id": "1",
    "last_name": "Montcenis"
  },
  {
    "id": "7",
    "username": "vivianeglz",
    "first_name": "Viviane",
    "last_name": "Gonzalez",
    "email": "viviane.gonzalez@outlook.fr",
    "birthdate": "02/06/1995",
    "avatar": "kikou.png"
  },
  {
    "id": "3",
    "username": "antoinesinety",
    "first_name": "Antoine",
    "last_name": "de Sinety",
    "email": "fake@yopmail.fr",
    "birthdate": "08/06/1995",
    "avatar": "Image.png"
  },
  {
    "id": "4",
    "username": "tgauzin",
    "first_name": "Thomas",
    "last_name": "Gauzin",
    "email": "thomas.gauzin@gmail.com"
  }
]
