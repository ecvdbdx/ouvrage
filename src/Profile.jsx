import React from 'react';
import moment from 'moment';
import "isomorphic-fetch";

const EditBtn = props => (<button onClick={props.handleClick}>Modifier</button>);

export default class Profile extends React.Component {
  constructor (props) {
    super(props);

    this.state = {
      edit: false,
      birthdate: moment(this.props.birthdate).format('YYYY-MM-DD'),
      githubData: [],
      githubRepoData: []
    };

    this.handleUpdating = this.handleUpdating.bind(this);
  }

  componentDidMount ()
  {
    fetch('https://api.github.com/users/' + this.props.student.username)
      .then(resp => {
        if(resp.ok) {
          resp.json().then(res => {
            console.log(res);
            this.setState({
              githubData: res
            });
          });
        } else {
          console.log('Mauvaise réponse du réseau');
        }
      })
      .catch(err => console.log(`ERROR in fetchJson : ${err}`));
    fetch('https://api.github.com/users/' + this.props.student.username + '/repos')
      .then(resp => {
        if(resp.ok) {
          resp.json().then(res => {
            console.log(res[0]);
            this.setState({
              githubRepoData: res
            });
          });
        } else {
          console.log('Mauvaise réponse du réseau');
        }
      })
      .catch(err => console.log(`ERROR in fetchJson : ${err}`));
  }

  handleUpdating () {
    this.setState({ edit: !this.state.edit });
  }

  handleSubmit (event) {
    event.preventDefault();
    let firstName = event.target.elements.first_name.value,
      lastName = event.target.elements.last_name.value,
      username = event.target.elements.username.value,
      email = event.target.elements.email.value,
      birthdate = event.target.elements.birthdate.value,
      idStudent = event.target.elements.id_student.value;
    console.log('Edit asked. Values : ');
    console.log('firstname : ' + firstName);
    console.log('lastname : ' + lastName);
    console.log('username : ' + username);
    console.log('email : ' + email);
    console.log('birthdate : ' + birthdate);
    console.log('idStudent : ' + idStudent);
  }

  render () {
    const { id, firstName, lastName, username, email, avatar } = this.props.student;

    return (
      <div id="show-profile">
        <div className="wrap-header">
          <h1 className="title">{firstName} {lastName}</h1>
          <EditBtn handleClick={this.handleUpdating} />
        </div>
        {this.state.edit ? (
          <div className="wrap-content">
            <div className="avatar">
              <img src={avatar} alt="avatar"/>
            </div>
            <div className="content">
              <form onSubmit={this.handleSubmit}>
                <input name="id_student" type="hidden" value={id}/>
                <div className="wrap-input">
                  <label>Prénom</label>
                  <input type="text" id="first_name" name="first_name" defaultValue={firstName}/>
                </div>
                <div className="wrap-input">
                  <label>Nom de famille</label>
                  <input type="text" id="last_name" name="last_name" defaultValue={lastName}/>
                </div>
                <div className="wrap-input">
                  <label>Username</label>
                  <input type="text" id="username" name="username" defaultValue={username}/>
                </div>
                <div className="wrap-input">
                  <label>Adresse e-mail</label>
                  <input type="email" id="email" name="email" defaultValue={email}/>
                </div>
                <div className="wrap-input">
                  <label>Date de naissance</label>
                  <input type="date" id="birthdate" name="birthdate" defaultValue={this.state.birthdate}/>
                </div>
                <input type="submit" value="Submit"/>
              </form>
            </div>
          </div>
        ) : (
          <div className="wrap-content">
            <div className="wrap-datas">
              <div className="avatar">
                <img src={avatar ? avatar : this.state.githubData.avatar_url} alt="avatar"/>
              </div>

              <div className="github">
                <h2 className="subtitle">Profil Github</h2>
                <p><span className="key">Login : </span>{this.state.githubData.login}</p>
                <p><span className="key">Created at : </span>{moment(this.state.githubData.created_at).format('DD-MM-YYYY')}</p>
                <p><span className="key">Followers : </span>{this.state.githubData.followers}</p>
                <p><span className="key">Following : </span>{this.state.githubData.following}</p>
                <p><span className="key">Public repos : </span>{this.state.githubData.public_repos}</p>
                <ul>
                  {/*<li>{this.state.githubRepoData.0.name}</li>*/}
                </ul>
              </div>
            </div>
            <div className="content">
              <h2 className="subtitle">Profil personnel</h2>
              <p>{firstName}  {lastName}</p>
              <p>{username}</p>
              <p>{email}</p>
              <p>{this.state.birthdate}</p>
            </div>
          </div>
        )}
      </div>
    );
  }
}
