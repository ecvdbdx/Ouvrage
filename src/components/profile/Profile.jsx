import React from 'react';
import moment from 'moment';
import { getIdAirTable, destroyUser } from '../../utils/airtable';

const EditBtn = props => (<button onClick={props.handleClick}>Modifier</button>);

export default class Profile extends React.Component {
  constructor (props) {
    super(props);

    this.state = {
      edit: false,
      birthdate: moment(this.props.birthdate).format('YYYY-MM-DD')
    };

    this.handleUpdating = this.handleUpdating.bind(this);
  }

  handleUpdating () {
    this.setState({ edit: !this.state.edit });
  }

  handleSubmit (event) {
    event.preventDefault();
    let firstName = event.target.elements.first_name.value,
      lastName = event.target.elements.last_name.value,
      username = event.target.elements.username.value,
      email = event.target.elements.email.value,
      birthdate = event.target.elements.birthdate.value,
      idStudent = event.target.elements.id_student.value;
    console.log('Edit asked. Values : ');
    console.log('firstname : ' + firstName);
    console.log('lastname : ' + lastName);
    console.log('username : ' + username);
    console.log('email : ' + email);
    console.log('birthdate : ' + birthdate);
    console.log('idStudent : ' + idStudent);
  }

  render () {
    const { id, firstName, lastName, username, email, avatar } = this.props.student;

    return (
      <div>
        <div className="wrap-header">
          <h1 className="title">{firstName} {lastName}</h1>
          <EditBtn handleClick={this.handleUpdating} />
          <button onClick={() => getIdAirTable(id, destroyUser)}>Supprimer</button>
        </div>
        {this.state.edit ? (
          <div className="wrap-content">
            <div className="avatar">
              <img src={avatar} alt="avatar"/>
            </div>
            <div className="content">
              <form onSubmit={this.handleSubmit}>
                <input name="id_student" type="hidden" value={id}/>
                <div className="wrap-input">
                  <label>Pr√©nom</label>
                  <input type="text" id="first_name" name="first_name" defaultValue={firstName}/>
                </div>
                <div className="wrap-input">
                  <label>Nom de famille</label>
                  <input type="text" id="last_name" name="last_name" defaultValue={lastName}/>
                </div>
                <div className="wrap-input">
                  <label>Username</label>
                  <input type="text" id="username" name="username" defaultValue={username}/>
                </div>
                <div className="wrap-input">
                  <label>Adresse e-mail</label>
                  <input type="email" id="email" name="email" defaultValue={email}/>
                </div>
                <div className="wrap-input">
                  <label>Date de naissance</label>
                  <input type="date" id="birthdate" name="birthdate" defaultValue={this.state.birthdate}/>
                </div>
                <input type="submit" value="Submit"/>
              </form>
            </div>
          </div>
        ) : (
          <div className="wrap-content">
            <div className="avatar">
              <img src={avatar} alt="avatar"/>
            </div>
            <div className="content">
              <p>{firstName}  {lastName}</p>
              <p>{username}</p>
              <p>{email}</p>
              <p>{this.state.birthdate}</p>
            </div>
          </div>
        )}
      </div>
    );
  }
}
