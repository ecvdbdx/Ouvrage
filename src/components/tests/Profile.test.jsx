import React from 'react';
import Profile from './../profile/Profile';
import renderer from 'react-test-renderer';
import students from './../../../mocks/students.json';

it('renders correctly', () => {
  const tree = renderer
    .create(<Profile student={students[2]}/>)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
