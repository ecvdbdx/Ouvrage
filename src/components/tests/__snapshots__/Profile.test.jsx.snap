// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div>
  <div
    className="wrap-header"
  >
    <h1
      className="title"
    >
      Vincent
       
      Rodriguez
    </h1>
    <button
      onClick={[Function]}
    >
      Modifier
    </button>
    <button
      onClick={[Function]}
    >
      Supprimer
    </button>
  </div>
  <div
    className="wrap-content"
  >
    <div
      className="avatar"
    >
      <img
        alt="avatar"
        src="https://s3.amazonaws.com/mixture-mixed/21975/17188/img/photos/portraits/dogs/golden-retriever.jpg"
      />
    </div>
    <div
      className="content"
    >
      <p>
        Vincent
          
        Rodriguez
      </p>
      <p>
        rodrigodelanoche
      </p>
      <p>
        vincentrodrigu@hotmail.fr
      </p>
      <p>
        2018-06-07
      </p>
    </div>
  </div>
</div>
`;
